server/src/test/java/org/opensearch/action/search/AbstractSearchAsyncActionTests.java

```diff
@@ -528,9 +603,346 @@ public void onFailure(Exception e) {
         assertThat(searchResponse.getSuccessfulShards(), equalTo(shards.length));
     }
 
+    public void testSearchRequestListeners() {
+        SearchCoordinatorStatsTesting searchCoordinatorStatsTesting = new SearchCoordinatorStatsTesting();
```

Comment:
Can't we reuse the original object instead of using this? Also the naming could be better, something like SearchCoordinatorMockStats